### ISSUE TITLE
encoded string format

### ISSUE THREAD

**Issue by fdl333** [josephpal/esp32-Encrypt#2]
First of all thank you for wonderful work!

NO issues at all (yet) but a question:

I noticed that the encryped string consists of many unreadable characters. I use aes-128 encoding a lot in php and ret a more "printable" string.
I will be sending the encoded string via mqtt and saving it to db.
I will also be decoding this string in javacript (both in node and web frontend)

Any guidance on this pls?

**Comment by josephpal** [comment]
> First of all thank you for wonderful work!
> 
> NO issues at all (yet) but a question:
> 
> I noticed that the encryped string consists of many unreadable characters. I use aes-128 encoding a lot in php and ret a more "printable" string.
> I will be sending the encoded string via mqtt and saving it to db.
> I will also be decoding this string in javacript (both in node and web frontend)
> 
> Any guidance on this pls?

oh man i am soooooo sorry, i totally missed your issue :( ....i recognized it right now, wtf
The problem about making something human readable belongs only to us as humans i would say ;) If something is not printable at all means that it is not interpretable by the encoding format the current editor/console or what so ever is using. For example, if you print the string in a hex format, you will get every character represented by one byte (e.g. '=' is 0x**3D** using standard ASCII).

To sum it up, as long as you know in which format / type your information is encrypted, you don't have to worry about ;)
Hope that helps a little bit.

**Comment by josephpal** [comment]
if you want, you can take a look inside the [documentation](https://github.com/josephpal/esp32-Encrypt/blob/master/documentation/Cipher-class-explanation.pdf) folder, there are some useful information about the problem representing the result human readable as characters or in HEX ;)

**Comment by DanMH17** [comment]
Hi Joseph, I´ve been working with this, thanks a lot. But I have a question, How can I get the hex format of the encrypt string?

**Comment by josephpal** [comment]
> Hi Joseph, I´ve been working with this, thanks a lot. But I have a question, How can I get the hex format of the encrypt string?

Hi, 

maybe you want to take a look at: https://www.arduino.cc/reference/en/language/functions/communication/serial/print/
As i mentioned before, the hex format is only a possible representation of a variable. In your case: 

`Serial.println(encryptedString, HEX)` 

should work.

Greetings
